### PR TITLE
Copter: proper DO_MOUNT_CONTROL

### LINKED ISSUE #2664: Copter: Change mount_mode with DO_MOUNT_CONTROL
DO_MOUNT_CONTROL should consume parameter 7 and change the mount_mode.  That way, arducopter can do a DO_MOUNT_CONTROL in guided mode, not just auto.

### PR DESCRIPTION
**Pull request by squilter**
Fixes #2664

### CODE REVIEW

**Comment by squilter** [comment]
@rmackay9 would you mind taking a look at this one when you have time?

**Comment by rmackay9** [comment]
@squilter, yup.  definitely haven't forgotten.  will look soon.

**Comment by squilter** [comment]
Thanks

**Comment by rmackay9** [comment]
Merged with some changes.  Thanks!

In particular I removed the new support for the DO_MOUNT_CONFIGURE sent as a command long because upon further reflection, I couldn't think of why someone would use this instead of using he DO_MOUNT_CONTROL message which also allows setting the mode.  The only other thing that do-mount-configure offered was changing which axis are stabilized which I don't think is actually a useful feature because these are just 2 of about 20 parameters that need to be set to configure a gimbal.

File: libraries/AP_Mount/AP_Mount_Backend.cpp [hunk 1]
```diff
@@ -26,27 +26,40 @@ void AP_Mount_Backend::set_roi_target(const struct Location &target_loc)
...
 {
     __mavlink_mount_control_t packet;
     mavlink_msg_mount_control_decode(msg, &packet);
 
+    control((int32_t) packet.input_a, (int32_t) packet.input_b, (int32_t) packet.input_c, 0);
+}
+
+void AP_Mount_Backend::control(int32_t pitch_or_lat, int32_t roll_or_lon, int32_t yaw_or_alt, uint8_t mount_mode)
+{
+    if(mount_mode > 0) {
```
Comment (squilter) [inline libraries/AP_Mount/AP_Mount_Backend.cpp]: @rmackay9 This line right here is why I decided that we needed to keep the `do_mount_configure`.  0 is a valid mount mode (`MAV_MOUNT_MODE_RETRACT`).  With this implementation, we wouldn't be able to set that.  We would need a do_mount_configure.  I recognize that this is probably not very important to ArduCopter, but it still seems like a crappy hack to have to use zero to ignore that field, when zero is actually supposed to mean something.

**Comment by squilter** [comment]
I'll do a similar thing for the digicam messages.

File: libraries/AP_Mount/AP_Mount_Backend.cpp (hunk 1 above)
Comment (squilter) [inline libraries/AP_Mount/AP_Mount_Backend.cpp]: Nevermind, I just saw you change that leaves the mode the same if the old message is used, but requires the mode to be set with the new message.  That seems like a good solution.

**Pull request closed without merging**